### PR TITLE
Refactor GSoc General Guidelines 

### LINKED ISSUE #44: Rephrase the contributors guidelines
**Is your feature request related to a problem? Please describe.**
- We need to rephrase the contributor guidelines a bit

**Describe the solution you'd like**
- Refactor the guidelines in a better way

https://handbook.sefglobal.org/gsoc-contributor-guidance

**Are you working on it?**
No

### PR DESCRIPTION
**Pull request by Dilini-457**
To suggest a change to the [Handbook](https://handbook.sefglobal.org/), please review the [Handbook Guidlines](https://handbook.sefglobal.org/organisation/handbook/usage#handbook-guidelines) section.

### Purpose:
The purpose of this PR is to fix #44

### Please provide a brief explanation for this change:

**Preview Link:**

https://pr-45-sef-handbook.surge.sh/

### Please indicate the types of revisions being suggested for the Handbook (please check all that apply):

* [ ] Small improvement (typos, clarifications, etc.)
* [ ] Adding a new section
* [ x ] Modifying existing section
* [ ] Documenting a new process
* [ ] Adding a new page or directory
* [ ] Other

### Author Checklist

* [ x ] Provided a concise title for the PR
* [ ] Provided a brief explanation for this change (Say why not just what)
  * (Attach screenshots, Slack conversations, etc. as necessary)
* [ ] Indicated the types of changes included in this PR
* [ ] Verified that no confidential data is in this PR
* [ ] Posted an update in [#handbook](https://sefheadquarters.slack.com/archives/C0376JVK2HF) channel
* [ ] If the changes affect team members, or warrant an announcement in another way, please consider posting an update in [#general](https://sefheadquarters.slack.com/archives/CJ89Y7JAV) channel

### CODE REVIEW

File: docs/gsoc-contributor-guidance.md [hunk 1]
```diff
@@ -9,42 +9,42 @@ Are you interested in participating in a Google Summer of Code project with SEF?
...
-3. Actively participate in SEF slack channel and SEF Hive.
-4. Introduce yourself on the introduction thread on SEF Hive.
-5. Contribute to SEF by working on github issues. Select issues from github, work on them, and submit pull requests to the respective repository.
-6. Identify potential issues in SEF projects. Create new github issues for unreported issues.
-7. Increase visibility on SEF Hive and Slack by helping others and engaging in discussions.
-8. Conduct code reviews. Reviewing others' code is essential for learning the SEF code base.
+1. Get acquainted with SEF and your chosen projects by referring to the Developer Guide and Getting Started as a Developer.
+2. Initiate your involvement by asking questions and exploring project ideas in the GSoC category on SEF Hive.
+3. Active participation in the SEF Slack channel and SEF Hive is crucial for community engagement.
+4. Introduce yourself in the dedicated thread on SEF Hive to connect with the community.
```
Comment (anjula-sack) [inline docs/gsoc-contributor-guidance.md]:
```suggestion
4. Introduce yourself in the [introduction](https://sef.discourse.group/t/welcome-please-introduce-yourself/7?u=anjisvj) thread on SEF Hive to connect with the community.
```

File: docs/gsoc-contributor-guidance.md [hunk 2]
```diff
@@ -9,42 +9,42 @@ Are you interested in participating in a Google Summer of Code project with SEF?
...
-1. Familiarize yourself with SEF and the project(s) you're interested in. Refer to the Developer Guide, Getting Started as a Developer, and engage with the community for any queries.
-2. Explore project ideas and ask questions in the GSoC category on SEF Hive.
-3. Actively participate in SEF slack channel and SEF Hive.
-4. Introduce yourself on the introduction thread on SEF Hive.
-5. Contribute to SEF by working on github issues. Select issues from github, work on them, and submit pull requests to the respective repository.
-6. Identify potential issues in SEF projects. Create new github issues for unreported issues.
-7. Increase visibility on SEF Hive and Slack by helping others and engaging in discussions.
-8. Conduct code reviews. Reviewing others' code is essential for learning the SEF code base.
+1. Get acquainted with SEF and your chosen projects by referring to the Developer Guide and Getting Started as a Developer.
+2. Initiate your involvement by asking questions and exploring project ideas in the GSoC category on SEF Hive.
```
Comment (anjula-sack) [inline docs/gsoc-contributor-guidance.md]:
```suggestion
2. Initiate your involvement by asking questions and exploring project ideas on SEF Hive.
```

File: docs/gsoc-contributor-guidance.md [hunk 3]
```diff
@@ -9,42 +9,42 @@ Are you interested in participating in a Google Summer of Code project with SEF?
...
 #### Before the Application
-1. Familiarize yourself with SEF and the project(s) you're interested in. Refer to the Developer Guide, Getting Started as a Developer, and engage with the community for any queries.
-2. Explore project ideas and ask questions in the GSoC category on SEF Hive.
-3. Actively participate in SEF slack channel and SEF Hive.
-4. Introduce yourself on the introduction thread on SEF Hive.
-5. Contribute to SEF by working on github issues. Select issues from github, work on them, and submit pull requests to the respective repository.
-6. Identify potential issues in SEF projects. Create new github issues for unreported issues.
-7. Increase visibility on SEF Hive and Slack by helping others and engaging in discussions.
-8. Conduct code reviews. Reviewing others' code is essential for learning the SEF code base.
+1. Get acquainted with SEF and your chosen projects by referring to the Developer Guide and Getting Started as a Developer.
```
Comment (anjula-sack) [inline docs/gsoc-contributor-guidance.md]:
```suggestion
1. Get acquainted with SEF and your chosen projects by referring to the [Getting Started](https://handbook.sefglobal.org/engineering-team/team) as a Developer.
```

**Review by anjula-sack — CHANGES_REQUESTED** [review]
Requested some minor changes @Dilini-457

**Comment by anjula-sack** [comment]
@Dilini-457 can you update the PR with the changes?

**Comment by anjula-sack** [comment]
@Dilini-457 can you update the PR with the changes?

**Comment by Dilini-457** [comment]
@anjula-sack Updated the PR with requested changes.

**Review by anjula-sack — APPROVED** [review]
Thank you! @Dilini-457

**Pull request merged**